src/library_async.js

```diff
@@ -649,24 +688,46 @@ mergeInto(LibraryManager.library, {
     },
 
     allocateData: function() {
-      // An asyncify data structure has two fields: the
-      // current stack pos, and the max pos.
-      var ptr = _malloc(Asyncify.StackSize + 8);
-      HEAP32[ptr >> 2] = ptr + 8;
-      HEAP32[ptr + 4 >> 2] = ptr + 8 + Asyncify.StackSize;
+      // An asyncify data structure has three fields:
+      //  0  current stack pos
+      //  4  max stack pos
+      //  8  id of function at bottom of the call stack (callStackIdToFunc[id] == js function)
+      //
+      // The Asyncify ABI only interprets the first two fields, the rest is for the runtime.
+      // We also embed a stack in the same memory region here, right next to the structure.
+      // This struct is also defined as asyncify_data_t in emscripten/fiber.h
+      var ptr = _malloc(12 + Asyncify.StackSize);
```

Comment:
I believe that these `12`s can be `C_STRUCTS.asyncify_data_s.__size__`

Reply:
Oh, I didn't know about `__size__`. I'll change that, thanks.